ERCS/erc-7683.md

```diff
@@ -98,79 +118,241 @@ struct Input {
 struct Output {
 	/// @dev The address of the ERC20 token on the destination chain
 	/// @dev address(0) used as a sentinel for the native token
-	address token;
+	bytes32 token;
 	/// @dev The amount of the token to be sent
 	uint256 amount;
 	/// @dev The address to receive the output tokens
-	address recipient;
+	bytes32 recipient;
 	/// @dev The destination chain for this output
-	uint32 chainId;
+	uint64 chainId;
+}
+```
+
+### Initiated event
+
+A compliant `Initiated` event MUST adhere to the following abi:
+
+```solidity
+/// @title FillInstructions type
+/// @notice Instructions to parameterize each leg of the fill
+/// @dev Provides all the origin-generated information required to produce a valid fill leg
+struct FillInstructions {
+	/// @dev The contract address that the order is meant to be settled by
+	uint64 destinationChainId;
+	/// @dev The contract address that the order is meant to be filled on
+	bytes32 destinationSettler;
+	/// @dev The data generated on the origin chain needed by the destinationSettler to process the fill
+	bytes originData;
 }
 
+/// @notice Signals that an order has been initiated
+/// @param orderId a unique order identifier within this settlement system
+/// @param fillInstructions fillInstructions instructions to parameterize each fill leg, with each element representing a single leg
+event Initiated(bytes32 indexed orderId, FillInstructions[] fillInstructions);
 ```
 
-### ISettlementContract interface
+### Settlement interfaces
 
-A compliant settlement contract implementation MUST implement the `ISettlementContract` interface:
+A compliant origin settler contract implementation MUST implement the `IOriginSettler` interface:
 
 ```solidity
-/// @title ISettlementContract
-/// @notice Standard interface for settlement contracts
-interface ISettlementContract {
+/// @title IOriginSettler
+/// @notice Standard interface for settlement contracts on the origin chain
+interface IOriginSettler {
+	/// @notice Initiates the settlement of a gasless cross-chain order on behalf of a user.
+	/// @dev To be called by the filler.
+	/// @dev This method must emit the Initiated event
+	/// @param order The GaslessCrossChainOrder definition
+	/// @param signature The user's signature over the order
+	/// @param fillerData Any filler-defined data required by the settler
+	/// @returns fillInstructions instructions to parameterize each fill leg, with each element representing a single leg
+	function initiateFor(GaslessCrossChainOrder order, bytes signature, bytes originFillerData) external returns (FillInstructions[]);
```

Comment:
I'm not fully sure I understand this function/naming. From the section below, I understood the gasless flow to not require any tx on the origin chain until after the intent is filled on the target chains. Is this not the case?

Reply:
It's not necessarily the case. The standard was originally built with the following gasless ordering in mind:
1. Filler gets user signed order.
2. Filler sends a transaction on the origin chain to escrow the user's tokens.
3. Filler sends a fill on the destination(s).

2 generally comes before 3 in the classic flow to protect the filler from the user moving their tokens after the fill but before the filler escrows the user's tokens.

Other flows where the user pre-locks their tokens wouldn't need this ordering. The standard fully supports that, but the "classic" ordering is the reason for the naming. Do you have an alternative name for initiate that would better communicate this flexibility?